ws-server/src/server.rs

```diff
@@ -365,6 +396,14 @@ impl Builder {
 		self
 	}
 
+	/// Register a new resource kind. Errors if `label` is already registered, or if number of
+	/// registered resources would exceed 8.
```

Comment:
Please document the meaning of `capacity` and `default`.

I wonder if having a `ResourceBuilder` with `.with_cap` and `.with_default` methods would make the api more self-explanatory? Not a blocker but one to ponder.